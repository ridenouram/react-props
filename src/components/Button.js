import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Button extends PureComponent {
  static propTypes = {
    color: PropTypes.string.isRequired
  }

  buttonStyle = {
    backgroundColor: this.props.color,
    width: '75px',
    height: '75px',
    margin: '10px'
  };

  clickHandler = event => {
    console.log(this.props.color);
  }

  render() {
    return (
      <button type="button" style={this.buttonStyle} onClick={this.clickHandler}>{this.props.color}</button>
    );
  }
}
