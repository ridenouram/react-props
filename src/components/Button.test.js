import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button Component', () => {
  it('snapshot', () => {
    const wrapper = shallow(<Button color="red"></Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
