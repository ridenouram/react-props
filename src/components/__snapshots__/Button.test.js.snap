// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component snapshot 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "height": "75px",
      "margin": "10px",
      "width": "75px",
    }
  }
  type="button"
>
  red
</button>
`;
